n3fit/src/n3fit/hyper_optimization/mongofiletrials.py

```diff
@@ -0,0 +1,290 @@
+            ]
+            if workdir:
+                args.extend(["--workdir", workdir])
+            if exp_key:
+                args.extend(["--exp-key", exp_key])
+            args.extend(["--poll-interval", str(poll_interval)])
+            if use_subprocesses:
+                args.append("--no-subprocesses")
+
+            # start the worker as a subprocess
+            try:
+                my_env = os.environ.copy()
+
+                if num_gpus_available:
+                    # set CUDA_VISIBLE_DEVICES environment variable
+                    # the GPU index assigned to each worker i is given by mod(i, num_gpus_available)
+                    my_env["CUDA_VISIBLE_DEVICES"] = str(i % num_gpus_available)
+                    # set tensorflow memory growth
+                    my_env["TF_FORCE_GPU_ALLOW_GROWTH"] = "true"
+                    # avoid memory fragmentation issues?
+                    # my_env["TF_GPU_ALLOCATOR"] = "cuda_malloc_async"
+
+                # run mongo workers
+                # we could use stdout=subprocess.DEVNULL and stderr=subprocess.DEVNULL in Popen to suppress output info
+                worker = subprocess.Popen(args, env=my_env)
+                self.workers.append(worker)
+                log.info(f"Started mongo worker {i+1}/{self.num_workers}")
+            except OSError as err:
+                msg = f"Failed to execute {args}. Make sure you have MongoDB installed."
+                raise EnvironmentError(msg) from err
+
+    def stop_mongo_workers(self):
+        """Terminates all active mongo workers."""
+        for worker in self.workers:
+            try:
+                worker.terminate()
+                worker.wait()
+                log.info(f"Stopped mongo worker {self.workers.index(worker)+1}/{self.num_workers}")
+            except Exception as err:
+                log.error(
+                    f"Failed to stop mongo worker {self.workers.index(worker)+1}/{self.num_workers}: {err}"
+                )
+
+    def compress_mongodb_database(self):
+        """Saves MongoDB database as tar file"""
+        # check if the database exist
+        if not os.path.exists(f"{self.db_name}" and not glob.glob('65*')):
+            raise FileNotFoundError(
+                f"The MongoDB database directory '{self.db_name}' does not exist. "
+                "Ensure it has been initiated correctly and it is in your path."
+            )
+        # create the tar.gz file
+        try:
+            log.info(f"Compressing MongoDB database into {self.database_tar_file}")
+            subprocess.run(
+                ['tar', '-cvf', f'{self.database_tar_file}', f'{self.db_name}'] + glob.glob('65*'),
+                check=True,
+            )
+        except subprocess.CalledProcessError as err:
+            raise RuntimeError(f"Error compressing the database: {err}")
+
+    @staticmethod
+    def extract_mongodb_database(database_tar_file):
```

Comment:
In the same way, you can use `tarfile` to untar the compressed database.
```python
import tarfile

with tarfile.open(path_to_tar) as tar:
    tar.extractall(dest)
```

Reply:
Done!